code/modules/shuttle/on_move.dm

```diff
@@ -93,6 +93,17 @@ All ShuttleMove procs go here
 	if(z == ZLEVEL_MINING) //Avoids double logging and landing on other Z-levels due to badminnery
 		SSblackbox.add_details("colonies_dropped", "[x]|[y]|[z]") //Number of times a base has been dropped!
 
+/obj/machinery/gravity_generator/main/beforeShuttleMove()
+	update_list() //make sure everything's correct before running
+	if(!on)
+		return
```

Comment:
update list does the same thing as the other code here, you just need to change the `on` var and then update_list()

in the `afterShuttleMove()` you can check charging_state for if it was on before, `if(charging_state == POWER_IDLE)` then set on to true.